x/msgfees/keeper/keeper.go

```diff
@@ -176,10 +176,12 @@ func (k Keeper) DeductFeesDistributions(bankKeeper bankkeeper.Keeper, ctx sdk.Co
 	if neg {
 		return sdkerrors.ErrInsufficientFunds.Wrapf("negative balance after sending coins to accounts and fee collector: remainingFees: %q, sentCoins: %q, distribution: %v", remainingFees, sentCoins, fees)
 	}
-	// sweep the rest of the fees to module
-	err := bankKeeper.SendCoinsFromAccountToModule(ctx, acc.GetAddress(), k.feeCollectorName, unsentFee)
-	if err != nil {
-		return sdkerrors.ErrInsufficientFunds.Wrap(err.Error())
+	if unsentFee.IsAllPositive() {
```

Comment:
Long story short, you should use `!unsentFee.IsZero()` for this test.

Long story:

Reading this, `unsentFee.IsAllPositive()` doesn't look right since `0hotdog,5stake` would return false. In that case `!unsentFee.IsZero()` would be better.

I did a quick test though, and (at least right now), it looks like `SafeSub` doesn't include entries that end up being 0. So this ends up working. But since that's not obvious, and depends on external functionality, a comment about why `IsAllPositive` is used would be helpful for future maintainers.

Then, all that being said, if `SafeSub` resulted in a negative amount, we've returned an error. Combine that with knowing that `SafeSub` doesn't return entry with `0` amounts. What you're left with is that either `unsentfee` is zero (has nothing in it), or it has one or more positive amounts in it. So again, it'd just be better to use `! .IsZero`.

Reply:
Yeah, good call.   I originally had it as that.